### PR TITLE
[GKD] Fix Liger fused JSD path computing wrong loss

### PR DESCRIPTION
**Pull request by roycho96**
## What

When `use_liger_kernel=True`, `GKDTrainer.compute_loss` was silently training against a different objective than the default path:

1. `LigerFusedLinearJSDLoss` defaults to `weight_hard_loss=0.5, weight_soft_loss=0.5`, so the loss was `0.5 * CE(student, true_labels) + 0.5 * JSD(student, teacher)` instead of the pure JSD that `generalized_jsd_loss` computes.
2. The kernel was constructed with `temperature=args.temperature`. `args.temperature` is the generation-sampling temperature (default `0.9`); the non-Liger path calls `generalized_jsd_loss(...)` without a `temperature` argument and therefore uses `1.0`. The two paths were scaling logits differently before computing JSD.

Both came in with #3946 ([GKD] add liger loss). `distillation_trainer.py` and `gold_trainer.py` already pass `weight_hard_loss=0.0, weight_soft_loss=1.0`; only GKD was missed.

## Reproducer

Same student + same teacher, batch from `trl-internal-testing/zen`, tiny Qwen2.5 model:

| path | loss |
|---|---|
| non-Liger `generalized_jsd_loss` | 0.297 |
| Liger (defaults: `0.5 * CE + 0.5 * JSD`, `temperature=args.temperature=0.9`) | 0.342 |
| Liger after fix (`pure JSD`, `temperature=1.0`) | 0.297 |

The 15% gap is the silent objective swap. With student == teacher (so JSD ≈ 0) the gap was `~6.0`, dominated by the unwanted `0.5 * CE` term.

## Fix

```python
self.liger_jsd_loss = LigerFusedLinearJSDLoss(
    beta=args.beta,
    ignore_index=-100,
    temperature=1.0,        # was args.temperature
    compiled=False,
    weight_hard_loss=0.0,   # added
    weight_soft_loss=1.0,   # added
)
```

## Tests

- New `test_liger_loss_matches_non_liger_loss` builds two trainers with `use_liger_kernel=False/True` on the same student/teacher and asserts `compute_loss` agrees within fp tolerance.
- Existing `test_compute_loss_return_outputs_with_liger` still passes.
- `test_gkd_trainer_with_liger` OOMs on a 16 GB card both before and after this PR (independent issue).

---

> [!NOTE]
> **Medium Risk**
> Changes the loss used when `use_liger_kernel=True`, which directly affects training behavior and model quality, but is scoped to the experimental GKD trainer path. Adds a targeted regression test to catch future mismatches between fused and reference loss computations.
> 
> **Overview**
> Fixes the `GKDTrainer` Liger-kernel loss path to match the default objective by configuring `LigerFusedLinearJSDLoss` as **pure JSD** (sets `weight_hard_loss=0.0`, `weight_soft_loss=1.0`) and removing temperature scaling differences.
> 
> Adds a new accelerator-gated test asserting that `compute_loss` with and without Liger produces matching losses on the same batch/weights, preventing silent objective drift.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit efa9aab698b7b325becd0064c8ede4498cdae41a. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by kashif** [comment]
thanks @roycho96 checking

File: trl/experimental/gkd/gkd_trainer.py [hunk 1]
```diff
@@ -143,11 +143,16 @@ def __init__(
         # Liger fused GKD loss (JSD)
         self.use_liger_gkd_loss = False
         if args.use_liger_kernel:
+            # Match the non-Liger path: pure JSD (no hard CE component) and no temperature
+            # scaling, since `generalized_jsd_loss` is called without a `temperature` argument
+            # and `args.temperature` is for generation sampling only.
```
Comment (kashif) [inline trl/experimental/gkd/gkd_trainer.py]:
```suggestion
            # scaling, since `generalized_jsd_loss` is called without a `temperature` argument.
```

File: trl/experimental/gkd/gkd_trainer.py [hunk 2]
```diff
@@ -143,11 +143,16 @@ def __init__(
...
             self.liger_jsd_loss = LigerFusedLinearJSDLoss(
                 beta=args.beta,
                 ignore_index=-100,
-                temperature=args.temperature,
+                temperature=1.0,
```
Comment (kashif) [inline trl/experimental/gkd/gkd_trainer.py]:
```suggestion
```

Comment (kashif) [reply trl/experimental/gkd/gkd_trainer.py]: since 1.0 is the default anyways

**Review by kashif — APPROVED** [review]
